docs/src/content/docs/guides/sidebar.mdx

```diff
@@ -138,7 +138,8 @@ The configuration above generates the following sidebar:
 
 Starlight can automatically generate a group in your sidebar based on a directory of your docs.
 This is helpful when you do not want to manually enter each sidebar item in a group.
-Pages will be sorted alphabetically by filename by default.
+
+By default, pages will be sorted according to the filename in your [`defaultLocale`](/reference/configuration/#defaultlocale). This order generally follows the native speakers' agreed-upon “alphabetical” arrangement. However, variations exist between locales and even within a locale, where the definition of “alphabetical” may differ for formal versus informal text or based on case sensitivity. The exact ordering that Astro uses is the one defined by the [ICU Collation](https://unicode-org.github.io/icu/userguide/collation/) of your default locale without any [customization](https://unicode-org.github.io/icu/userguide/collation/customization/).
```

Comment:
```suggestion
By default, pages will be sorted in alphabetical order according to the filenames in your [`defaultLocale`](/reference/configuration/#defaultlocale). 

:::tip
The definition of “alphabetical” may differ between locales or even within a locale where conventions may differ based on context or case sensitivity. Astro follows the order defined by the [ICU Collation](https://unicode-org.github.io/icu/userguide/collation/) of your default locale without any [customization](https://unicode-org.github.io/icu/userguide/collation/customization/).
:::
```
Was thinking something like this might work! (Haven't viewed a preview, just editing in place.) What do we think?

Reply:
There are some edge cases if the locale includes a variation, like Traditional European Spanish (es-ES-u-co-trad) which allow letter groups interfere with the ordering, so it is not alphabetical even by their own definition. For example, in Traditional European Spanish "cosas" is sorted before "chica" even though they sort "h" before "o".

That is why I replaced the sentence stating that it _is_ alphabetical.

But that may be too purist of me. And as Tim Peters said, practicality beats purity.

Reply:
Acknowledged, but I also think we've maybe given people *enough* of a heads up that "alphabetical" might not be what they expect, so we probably don't have to describe the specifics to that full extent here?

What do we think visually of the addition? Tip not too big/distracting?

Reply:
If we go with a tip I think we can leave the previous sentence on the same paragraph as before.

Reply:
I think it looks fine. It is at the end of the explanation and before the examples so it doesn't break the flow IMO.

In the same paragraph as before:
![image](https://github.com/withastro/starlight/assets/11063910/47c56071-1902-4d2a-a043-7bcda3fc0faf)